src/Adapter/Form/ChoiceProvider/TaxRuleGroupChoiceProvider.php

```diff
@@ -53,19 +56,33 @@ public function getChoices()
      */
     public function getChoicesAttributes(): array
     {
-        $attrs = [];
-        foreach ($this->getRules() as $rule) {
-            // Keep first one found
-            if (!empty($attrs[$rule['name']]['data-tax-rate'])) {
-                continue;
-            }
+        $address = new Address();
+        $address->id_country = (int) Context::getContext()->country->id;
 
-            $attrs[$rule['name']] = [
-                'data-tax-rate' => !empty($rule['rate']) ? $rule['rate'] : 0,
+        $tax_rates = [];
+        foreach ($this->getRules() as $rule) {
+            $id_tax_rules_group = (int) $rule['id_tax_rules_group'];
+            $tax_calculator = TaxManagerFactory::getManager($address, $id_tax_rules_group)->getTaxCalculator();
+            $tax_rates[$id_tax_rules_group] = [
+                'id_tax_rules_group' => $id_tax_rules_group,
+                'rates' => [],
+                'computation_method' => (int) $tax_calculator->computation_method,
             ];
```

Comment:
Are you sure these data are relevant in the choices attributes? Besides I don't think they are correctly indexed because the array index should map the one used in `getChoices` so it should rather be `$tax_rates[$rule['name']]` (like you did in L75

This data will, at best, be ignored and at worst add some unexpected extra fields in the selector. If these fields are not used in the page after (unlike the `data-tax-rate` data attribute) then they shouldn't be part of the returned attributes. If they are not relevant and since they are not used in the following code maybe this should simply be removed.